### PR TITLE
Fix a BUG in ReportGateway and new report test files 

### PR DESCRIPTION
**Pull request by Allof02**
I didn't realize that Map can only contain unique keys. Since a Review can be reported by mutiple reporters, simple Map and HashMap won't work and will destory the Database

I added a MultiMap (from Java Guava) so I can store multiple Reports to an identical ReviewID

Mates who need to use my FileReportHistory pls update to the newest version of it;

There is NO changes to FileReportHistory constructor, so it wont affect your codes

Added more tests as well

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -9,7 +9,6 @@ repositories {
 dependencies {
     implementation 'junit:junit:4.13.1'
```
Comment (foreverwoods) [inline build.gradle:10]: Make sure this build.gradle is up to date with TIm's pull request gradle files and that it still works. There was a junit version change I believe

**Review by foreverwoods — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (prajogt) [reply build.gradle:10]: Yeah this should be changed to 4.13.2

**Review by prajogt — COMMENTED** [review]
We might have some conflicts with ReportTest, but if you approved David's request it should be ok.

Additionally pull the newest version to get the latest gradle updates

**Comment by Allof02** [comment]
im progressively move each unit test in ReportTest to its own test file so conflict doesnt really matter; I gonna delete ReportTest as soon as i finish other tests @prajogt

**Review by foreverwoods — APPROVED** [review]
Made approval yesterday, all that changed was the switch in junit version. Looks good.

**Review by prajogt — APPROVED** [review]
Ok works out, looks good

**Comment by Allof02** [comment]
Add more ReportTest so currenly have 89% line coverage for report system

**Review by foreverwoods — DISMISSED** [review]
All you added were tests, and the import change looked to just be IntelliJ's change, looks fine as long as it runs.

**Review by foreverwoods — APPROVED** [review]
Minimal change, approved.

**Comment by Allof02** [comment]
change button name in ReportScreen

**Review by JarodGratton — APPROVED** [review]

File: src/test/java/report_use_case/interactors/ExcaliburTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        User exUser = new User("TestUser", "1234567");
+        Review exReview = new Review("1", 5, "TestReview", "TestUser",
+                "TestRestaruantAddr");
+
+
+        //Review will be set to invisible if it has >= 10 reports
+        for (int i = 0; i <=15; i++){
+            exReview.addReport();
+        }
```
Comment (prajogt) [inline src/test/java/report_use_case/interactors/ExcaliburTest.java:23]: Maybe try doing edge cases, aka i = 10 and not 15

File: src/test/java/report_use_case/interactors/ExcaliburTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        User exUser = new User("TestUser", "1234567");
+        Review exReview = new Review("1", 5, "TestReview", "TestUser",
+                "TestRestaruantAddr");
+
+        //User will be banned if it has >= 30 reports
+        for (int i = 0; i <= 35; i++){
+            exUser.addReport();
+        }
+
```
Comment (prajogt) [inline src/test/java/report_use_case/interactors/ExcaliburTest.java:42]: Same edge case note here

**Review by prajogt — APPROVED** [review]
Looks good, just your bug fix and new tests as mentioned.
